### PR TITLE
Add jsons:parse() method to get a JSON from a string

### LINKED ISSUE #2839: string to json conversion issues
I have a json object stored in a DB as text/varchar

I read it to ballerina like 
```
datatable dt = sql:ClientConnector.select(foodlog, "select * from log ORDER BY id DESC LIMIT ?", params);
var logObj,err = <json>dt;
```

when I print ```logObj.myjsonobject``` I get quotes escaped string like

```
{\"breakfast\":{\"item\":\"bigmac\",\"calories\":562.83},\"lunch\":{\"item\":\"rice\",\"calories\":205.4},\"dinner\":{\"item\":\"pizza\",\"calories\":2268.98},\"total_calories\":3037.21,\"date\":\"Jul 2 2017\"}
```
there is no way for me to convert this directly to a json object.

### PR DESCRIPTION
**Pull request by SupunS**
Fix #2839

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BJSON.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public BJSON(String jsonString, String schema) {
                 this.schema = OBJECT_MAPPER.readTree(schema);
             }
         } catch (Throwable t) {
-            handleJsonException("failed to create json: ", t);
+            handleJsonException(null, t);
```
Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BJSON.java]: Why 'null' here?

File: modules/ballerina-native/src/main/ballerina/ballerina/lang/jsons/natives.bal [hunk 2]
```diff
@@ -68,3 +68,6 @@ native function toString (json j) (string);
 @doc:Param { value:"newKey: The new key value to use" }
 native function rename (json j, string jsonPath, string oldKey, string newKey);
 
+@doc:Description { value:"Parses and gets a JSON from a string."}
```
Comment (sameera-jayasoma) [inline modules/ballerina-native/src/main/ballerina/ballerina/lang/jsons/natives.bal]: You can reword this description as follows.

"Parses the string argument as a JSON value"

Comment (SupunS) [inline modules/ballerina-native/src/main/ballerina/ballerina/lang/jsons/natives.bal]: Fixed.

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BJSON.java (hunk 1 above)
Comment (SupunS) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BJSON.java]: Here passing a message causes to log duplicate information in an error. e.g: `failed to parse json: failed to create json: ...` . hence only propagating the actual error message only. 

Fixed the null passing, by overloading the method.

**Pull request merged**